src/cgame/cg_view.cpp

```diff
@@ -922,6 +921,15 @@ static int CG_CalcFov()
 		}
```

Comment:
This is out of place now. I guess the follow and unfollow cases could be unified since it is the same command either way.

Reply:
It's the same command (it's togglable), but with this design, follow starts with middle mouse and stops with mouse right (cancel).

This is to make possible to cycle between 1st person/chase view while never stopping to follow a given player. Before, once you have switched from 1st person to chase, it was not possible to return to 1st person view without unfollowing entirely, restarting to follow, and recycling to find the player again.

Reply:
Note: we may add non-togglable commands (to start and stop following) to make the code explicit and remove some test that only exists because the command is a togglable command. But that's out of topic.

Reply:
I did not realize this is intentional. I don't like this change - attack3 is too obscure to use for the most important control when spectating.

Reply:
I did not chose or changed anything about the fact `attack3` starts to follow. I made `attack2` cancels the following.

On a side note, the alternative to `attack3` to start to follow would be `attack2` but then you would return to the first issue when you can't come back to 1st person view to chase or you would cancel with `attack3` which is not consistent with the builder cancel, and this time, absolutely not intuitive. The other alternative to start follow would be `attack1` which is something I am strongly against, because that means we can't join a game with attack1. Some games uses `space` (like xonotic) to join because `attack1` is used to follow, that means a spectator cannot fly up when free flying (and if the user mistakenly presses space by muscle reflex, he joins and lose his position, which is very frustrating. That alternative is so frustrating I developed a special phobia to press space when spectating in other games even ours.

So, I did not chose `attack3` to start to follow. That always worked like this. What I did was to move the follow cancel on `attack2` like the builder cancel (which is intuitive). Also when a player clicks `attack1` to join it also stops to follow.

This implementation looks to be the best implementation: everything is available on the mouse, that does not prevent to freefly. I would like to see that in other games instead of developing phobias. I just made it better by removing the “stop following” command from the follow view mode cycle.

Reply:
I like attack2 for start following. Then we have attack2 for cycling following vs. free-flying, and attack3 for cycling view mode. To me this is a good kind of consistency. If we do that I would be very happy with how it turns out, since it keeps the nice property of being able to use all following features with only the mouse, while necessitating the awkward mouse3 for only the least important operation.

Reply:
done, it works well.